config/config_partial.go

```diff
@@ -143,8 +143,13 @@ func PartialParseConfigFile(
 	include *IncludeConfig,
 	decodeList []PartialDecodeSectionType,
 ) (*TerragruntConfig, error) {
+	// Boolean to control if we're trying to parse a DependencyBlock.
+	// If true, we don't want to return if there is an error when trying to read the file
+	// as the dependency can be disabled
+	isDependencyBlock := decodeList[0] == DependencyBlock
+
 	configString, err := util.ReadFileAsString(filename)
-	if err != nil {
+	if err != nil && !isDependencyBlock {
```

Comment:
I was wondering why shouldn't be returned error in case if the first element from `decodeList` is DependencyBlock? 

I also tried to comment this part and added tests(TestOptionalDependenciesAreSkipped / TestDisabledDependenciesIsParsed) continued to be passing

Reply:
Hi @denis256, first sorry for the delay, I've been pretty busy lately. 

I added this condition because if a dependency block is disabled, potentially there isn't any terragrunt configuration at path `filename`.

I just checked what you said about tests, and indeed tests are passing without this condition. But, if I build the binary and use it in a real use case, `terragrunt` is exiting because it can't find a `terragrunt.hcl` file into the optional dependency location. For example, I have the following structure : 

```
mymockapp/
    module-a/
    module-b/
       terragrunt.hcl
    module-c/
       terragrunt.hcl
```

There isn't any configuration at path `module-a`. `module-b/terragrunt.hcl` is empty, and below the content of `module-c/terragrunt.hcl` : 

```
dependency "b" {
  config_path = "../b"
}

dependency "a" {
  enabled     = false
  config_path = "../a"
}
```

Now I apply the the `module-c` using the following command : 

```
terragrunt apply --terragrunt-working-dir=module-c
```

And I have the following output : 

```
ERRO[0000] Error reading file at path a/terragrunt.hcl: open a/terragrunt.hcl: no such file or directory 
ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1 
```

This is the current behavior and for sure I agree with it because before optional dependencies, we were assuming that every dependency has a `terragrunt.hcl` file. But now, I want to prevent dependencies parsing to return errors because they can be optional or not, and potentially they didn't have any `.hcl` files, like in the above example. 

But here the issue is probably a divergent point of view about optional dependencies. For me, and optional dependencies could not exists on the disk, or it can exists but shouldn't be taken in consideration during the parsing.

Are you ok with that ?  Depending on what you think about it, I'll remove this condition and the PR should be ready.

Thanks !